nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/PropertyTrashService.java

```diff
@@ -114,20 +123,49 @@ protected void doTrashDocument(DocumentModel doc) {
         }
         session.setDocumentSystemProp(docRef, SYSPROP_IS_TRASHED, Boolean.FALSE);
         notifyEvent(session, DOCUMENT_UNTRASHED, docForEvent);
+        if (processChildren && session.hasChildren(doc.getRef())) {
+            setTrashPropertyToChildren(doc, Boolean.FALSE);
+        }
 
         Set<DocumentRef> docRefs = new HashSet<>();
         docRefs.add(docRef);
 
         // now untrash the parent if needed
         if (parentRef != null && session.isTrashed(parentRef)) {
             DocumentModel parent = session.getDocument(parentRef);
-            parent.putContextData(SKIP_CHILDREN_PROCESSING_KEY, Boolean.TRUE);
-            Set<DocumentRef> ancestorDocRefs = doUntrashDocument(parent);
+            Set<DocumentRef> ancestorDocRefs = doUntrashDocument(parent, false);
             docRefs.addAll(ancestorDocRefs);
         }
         return docRefs;
     }
 
+    protected void setTrashPropertyToChildren(DocumentModel model, Boolean value) {
```

Comment:
javadoc

Reply:
it's not required as method is protected, do you think it's needed or simple enough ?

Reply:
I thought protected methods needed javadoc.